### PR TITLE
Update TypeScript behaviors and be more strict

### PR DESCRIPTION
**Pull request by kachick**
ref: https://github.com/tsconfig/bases/blob/d1543a94d47c3ef9af9987790742b09caf71b71d/bases/node16-strictest-esm.combined.json, https://github.com/mobu-of-the-world/mobu/pull/437#discussion_r892545520

In my understanding, Node 16 and modern browsers supports ES2021. So this will work 😇

### CODE REVIEW

**Comment by pankona** [comment]
naruhodo
https://www.cyokodog.net/blog/typescript-v3-8/

**Review by pankona — APPROVED** [review]

**Comment by pankona** [comment]
@kachick 

Approved! But I'm sorry to say that it seems there're some conflicts. Could you resolve them?

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "typescript.tsdk": "node_modules/typescript/lib"
+}
```
Comment (kachick) [inline .vscode/settings.json:3]: No endline, but this is generated by vscode itself

**Comment by kachick** [comment]
@pankona Thank you! I've resolve the conflicts and added vscode config to clarify which version of tsc should be used.

So I'll merge!

**Comment by pankona** [comment]
Thank you!

**Pull request merged**